setup.py

```diff
@@ -288,6 +315,124 @@ def readme():
+    _py_checker_command_line = [_py_checker, '--ignore=E265,W503']
+    _py_formatter_command_line = [_py_formatter, '--in-place', '--aggressive', '--aggressive']
+    _py_file_extensions = ['.py']
+
+    def _get_file_list(self, path, search_extentions):
+        files = []
+        exluded_directories_full_path = [os.path.join(path, excluded_dir)
+                                         for excluded_dir in self._project_directory_excluded]
+
+        # r=root, d=directories, f = files
+        for r, d, f in os.walk(path):
+            # match exclude pattern in current directory
+            found = False
+            for excluded_dir in exluded_directories_full_path:
+                if r.find(excluded_dir) >= 0:
+                    found = True
+
+            if found:
+                continue
+
+            for file in f:
+                filename, extention = os.path.splitext(file)
+                if extention in search_extentions:
+                    files.append(os.path.join(r, file))
+
+        return files
+
+    def initialize_options(self):
+        self.apply = 0
+
+    def finalize_options(self):
+        pass
+
+    def run(self):
+        root_dir = versioneer.get_root()
+        print("Project directory is: %s" % root_dir)
+
+        if self.apply:
+            self._c_formatter_command_line += ['-i']
+        else:
+            self._c_formatter_command_line += ['-output-replacements-xml']
+
+        import subprocess
+
+        bad_style_file_names = []
+
+        # C files handling
+        c_files = self._get_file_list(root_dir, self._c_file_extensions)
+        try:
+            for f in c_files:
+                command_output = subprocess.Popen(self._c_formatter_command_line + [f], stdout=subprocess.PIPE)
+                command_cout, command_cerr = command_output.communicate()
+                if not self.apply:
+                    if command_cout.find(b'<replacement ') > 0:
+                        bad_style_file_names.append(f)
+        except BaseException:
+            print("%s is not installed.\nPlease use: %s" % (self._c_formatter, self._c_formatter_install_msg))
+            exit(1)
+
+        # Python files handling
+        py_files = self._get_file_list(root_dir, self._py_file_extensions)
+        try:
+            for f in py_files:
+                if not self.apply:
+                    command_output = subprocess.Popen(
+                        self._py_checker_command_line + [f])
+                    returncode = command_output.wait()
+                    if returncode != 0:
+                        bad_style_file_names.append(f)
+                else:
+                    command_output = subprocess.Popen(
+                        self._py_formatter_command_line + [f])
+                    command_output.wait()
+        except BaseException:
+            print("%s is not installed.\nPlease use: %s" % (self._py_formatter, self._py_formatter_install_msg))
+            exit(1)
```

Comment:
this is a rather generic `except`, but the message is very specific. This can be very confusing if something else goes wrong. This should at least print the actual exception (message).
Using a python module rather than a pipe/process will also help with better error messages.